readme.md
###Project Details:
##Project Name - Memory Game
-The memory game is a indulging process that will stimulate your memory power. The game has several tiles generated on the screen. Each tile has an object that is hidden, among these hidden objects there are pairs. When an player clicks on a tile, the tile will flip and the object will be revealed. If the player successfully selects tiles with similar objects one after the other, a match will be made and the tiles will remain revealed for the rest of the game, otherwise both tiles will be flipped back.

-For my game all odd levels, such as 1,3,5..., are going to have pair of tiles similar chemical element its symbol and name.
-All the odd levels have the pairing tiles with symbol in one tile and name on the other. If the symbol and the name are a match, then a match is said to be succefully made.

-To win the game a player has to successfully match all the tiles in the game board.

##Main API:
#ShuffleTiles
-This API will shuffle the elments of an array and stores them in the same array for the defined length.

#startTimer
-This API runs the game timer and inserts the values into the gameTiles

#movesCounter
-This API keeps track of the no.of moves made in the game and add them on the HTML page.
-This API also decides the star rating based on the number of moves made for completion of level.
-The API also decides the comment for the user memory ability.

#levelEndSequence
-This API generates the end level sequence for the game after it is completed.
-This API also reset the variables for the game such as sec, minutes and other game variables.
-This API also generates button to move to next level or to replay the level.
-This API also checks if all the levels are cleared and generated the end game stats and text.

#flipTiles
-This flips the clicked tiles and reveals the tiles for the user.
-Then checks if two tiles are selected. If two tiles are selected then it runs a sequence of code to check for matching tiles.
-If matching then tiles are freezed, if not them they are flipped back.
-This also starts the movesCounter on each clicked.

#game
-This is a mojor API that creates all the game elements required for the game.
-It creates the game heading.
-It creates the game star elements.
-It creates the game timer div to add timer.
-It creates the move division to add moves of the game.
-It creates the game division and tiles.
-The shuffle array API is called and the tiles values are shuffled.
-The values are added to the tiles back face.
-The classes are added to the tiles.
-The tiles are revealed at the beginning of the game for a brief amount of time.
-The gameDiv length is adjusted based on the tiles length and screen size.
-The tiles are added with on click functionality.

#levelSelector
-The level is incremented after each level.
-The variables for ecah level are defined.
-14 levels are created in this set of insructions.
-The array with values is stored in a refrence array for odd levels and even levels.

##Minimum Requirements:
1.System installed with OS.
2.WEb browser compatible with HTML5.

#Installing
Open Index HTML document in web browser.


changelog.md
# Changelog
All notable changes to this project will be documented in this file.

### Date 17/05/2018
## Updated
# Created the game cards using Photoshop and updated it to the tiles.
# Player status decision system has been updated.
# Updated the animation for next and replay buttons.
# Created the background and updated it in the game.
# Major bugs have been eliminated in the game.


### Date 17/05/2018
## Updated
# The end game sequence is created.
# levels have been added to game.
# End level sequence is added.
# Created button for next level and replay level.

## To Do:
# Should update the background.
# Should update the game end sequence.
# Change colour of the tiles and board.

### Date 12/05/2018
## Updated
# Added audio files to the game.
# The score elements have been added.
# The stars have been added.
# The move elements are Added
# The timer element has been added.

## To Do:
# Should update the levels.
# Generate the end game menu.

### Date 07/05/2018
## Updated
# Research is made on animations to update the game.
# The flip animation has been added to the game tiles.
# The background image error was cleared by updating the image name in CSS. This was caused as the image name was changed after modifying the CSS.
# The setTimeout function has been implemented to delay the generation of new board. This clears the issue of the last tile not flipping.
# The game code has been modified to work with respect to the animation.

## To Do:
# The text has to disappear along with the flipping animation.
# The sounds have to be embedded into the game.
# The game levels have to be added to the game.

### Date 5/4/18
## Updated
# The set timer API is used to generate delay for the flip back functionality.
# The background color of game division is changed to #96c54d52.
# The background color of tile division for class list "open" is changed to #02b3e4.
# The background color of tile division for class list "close" is changed to #86d050.
# The text color of tile division for class list "open" is changed to #000000.
# The alphabets are replaced with chemical elements 'Hydrogen H', 'Helium He', 'Oxygen O', 'Carbon C', 'Neon Ne', 'Iron Fe', 'Nitrogen N', 'Silicon Si'.

## To Do:
# The last tile is not flipped when the game ends, have to correct end game sequence.
# add audio effects and more levels to games.
# add more levels to the game.

### Date 26/04/18
## updated
# created a flip and match function replacing match function to compare the array values.
# tiles are assigned values during flip and match function to correct comparing string.
# created id for tiles for assigning the flip back function.
# created two division classes open and close to create flip functionality.
# font attribute is only specified for Open to create flip effect.
# onClick function has been replaced with addEventListner to pass parameters to flip and match function.
# global variable memoryId is generated to store tile ids for flipback function.

## To Do:
# The flipback function running immediately without displaying cards.
# The visual effects are not effective for the text to display properly.
# Add audio effects and more levels to games.
# add more levels to the game

### Date 19/04/2018
## updated
# created a function to compare the array values.
# onClick function has been assigned to tiles to trigger match function.
# global variable memoryValue is generated to store elements for matching.

## To Do:
# The onClick function not triggering match function.
# method to hide and display text.
# Add audio effects and more levels to games.
# add more levels to the game

### Date 16/04/2018
## updated
# created a heading for the HTML file.
# added font Gloria Hallelujah to heading.
# added font size 40px for visibility.
# added a function to shuffle and store array elements.
# assigned the shuffled array elements to tiles using for loop.

## To Do:
# compare the tiles and check if matching.
# method to hide and display text.


### Date 12/04/2018
## updated
# added shadow to the game division for visual effect.
# added shadow to the tile division for visual effect.
# added values to the tiles and added font-size to display text.
# added text color #5950d0 to tile.

## To Do:
# Randomize the array elements.
# Compare the tiles.

###Date 04/04/2018
## updated
# centered the game division using margin: auto.
# tile division aligned using display: flex and flex-wrap: wrap CSS attributes.
# tile division shape contained using padding.
# added background color #716F71  to game division.
# added background color #141214 to tile division.

## To Do:
# Assign values to tiles.
# Compare the tiles.

## Updated:
# created game level 1 array with alphabets A,B,C,D,E,F,G,H.
# created ui java script for general functions.
# created division using ui.js functions.
# created tiles using ui.js functions.
# The script to create division from app.js has been removed.

## TO DO:
# add the game elements to the tiles.
# center the game division.
# align the tiles in game division.

### Date 25/03/2018
## Updated:
# Created file app.HTML, app.css, data files.
# Created game division.
# Created game tiles division.

## To do:
# center the division.
# center the tiles.
# align the tiles in division.
# add values to tiles.


js/ui.js
/* function to generate division*/
  function createDiv()
  {
    var div =  document.createElement("div");
    return div;
  }
/*function to create heading*/
  function createHeading(tiltleText){
    var element = document.createElement("h1");
    document.body.appendChild(element);
    var textNode = document.createTextNode(tiltleText);
    element.appendChild(textNode);
  }
//function to create paragraph
  function createParagraph(tiltleText){
    var element = document.createElement("p");
    var textNode = document.createTextNode(tiltleText);
    element.appendChild(textNode);
    return element;
  }
//function to create button
  function createButton(buttonText){
    console.log("Creating a Button: " + buttonText);
     var element = document.createElement("BUTTON");
     var span = document.createElement("span")
     var text = document.createTextNode(buttonText);
     span.appendChild(text);
     element.appendChild(span);
     return element;
  }
//function to create audio
function createAudio(AudioUrl){
  var element = new Audio(AudioUrl);
  return element;
}


data/gameData.js
/*All the game tile elements are stored in the memoryTilesArray.
Forr the first level we have few chemical elements alang with their symbols from periodic table.*/

// Storing names and Raw for the card
window.gameTiles =
[
  [
  {
    'name': 'H',
    'img': 'Raw/HYDROGEN/H1.png'
  },
  {
    'name': 'H',
    'img': 'Raw/HYDROGEN/H1.png'
  },
  {
    'name': 'He',
    'img': 'Raw/HELIUM/He1.png'
  },
  {
    'name': 'He',
    'img': 'Raw/HELIUM/He1.png'
  },
  {
    'name': 'O',
    'img': 'Raw/Oxygen/O1.png'
  },
  {
    'name': 'O',
    'img': 'Raw/Oxygen/O1.png'
  },
  {
    'name': 'C',
    'img': 'Raw/Carbon/C1.png'
  },
  {
    'name': 'C',
    'img': 'Raw/Carbon/C1.png'
  },
  {
    'name': 'Ne',
    'img': 'Raw/Neon/Ne1.png'
  },
  {
    'name': 'Ne',
    'img': 'Raw/Neon/Ne1.png'
  },
  {
    'name': 'Si',
    'img': 'Raw/Silicon/Si1.png'
  },
  {
    'name': 'Si',
    'img': 'Raw/Silicon/Si1.png'
  },
  {
    'name': 'Fe',
    'img': 'Raw/Iron/Fe1.png'
  },
  {
    'name': 'Fe',
    'img': 'Raw/Iron/Fe1.png'
  },
  {
    'name': 'N',
    'img': 'Raw/Nitrogen/N1.png'
  },
  {
    'name': 'N',
    'img': 'Raw/Nitrogen/N1.png'
  },
  {
    'name': 'Al',
    'img': 'Raw/Aluminium/Al1.png'
  },
  {
    'name': 'Al',
    'img': 'Raw/Aluminium/Al1.png'
  },
  {
    'name': 'Mg',
    'img': 'Raw/Magnesium/Mg1.png'
  },
  {
    'name': 'Mg',
    'img': 'Raw/Magnesium/Mg1.png'
  },
  {
    'name': 'Ca',
    'img': 'Raw/Calcium/Ca1.png'
  },
  {
    'name': 'Ca',
    'img': 'Raw/Calcium/Ca1.png'
  },
  {
    'name': 'S',
    'img': 'Raw/Sulphur/S1.png'
  },
  {
    'name': 'S',
    'img': 'Raw/Sulphur/S1.png'
  }
],[
{
  'name': 'H',
  'img': 'Raw/HYDROGEN/H2.png'
},
{
  'name': 'H',
  'img': 'Raw/HYDROGEN/H3.png'
},
{
  'name': 'He',
  'img': 'Raw/HELIUM/He2.png'
},
{
  'name': 'He',
  'img': 'Raw/HELIUM/He3.png'
},
{
  'name': 'O',
  'img': 'Raw/Oxygen/O2.png'
},
{
  'name': 'O',
  'img': 'Raw/Oxygen/O3.png'
},
{
  'name': 'C',
  'img': 'Raw/Carbon/C2.png'
},
{
  'name': 'C',
  'img': 'Raw/Carbon/C3.png'
},
{
  'name': 'Ne',
  'img': 'Raw/Neon/Ne2.png'
},
{
  'name': 'Ne',
  'img': 'Raw/Neon/Ne3.png'
},
{
  'name': 'Si',
  'img': 'Raw/Silicon/Si2.png'
},
{
  'name': 'Si',
  'img': 'Raw/Silicon/Si3.png'
},
{
  'name': 'Fe',
  'img': 'Raw/Iron/Fe2.png'
},
{
  'name': 'Fe',
  'img': 'Raw/Iron/Fe3.png'
},
{
  'name': 'N',
  'img': 'Raw/Nitrogen/N2.png'
},
{
  'name': 'N',
  'img': 'Raw/Nitrogen/N3.png'
},
{
  'name': 'Al',
  'img': 'Raw/Aluminium/Al2.png'
},
{
  'name': 'Al',
  'img': 'Raw/Aluminium/Al3.png'
},
{
  'name': 'Mg',
  'img': 'Raw/Magnesium/Mg2.png'
},
{
  'name': 'Mg',
  'img': 'Raw/Magnesium/Mg3.png'
},
{
  'name': 'Ca',
  'img': 'Raw/Calcium/Ca2.png'
},
{
  'name': 'Ca',
  'img': 'Raw/Calcium/Ca3.png'
},
{
  'name': 'S',
  'img': 'Raw/Sulphur/S2.png'
},
{
  'name': 'S',
  'img': 'Raw/Sulphur/S3.png'
}
]];


js/app.js
/* This is the file that is going to load for the game on to the webpage*/

//These are the global variable that are going to used for the game
var memoryTilesArray = [];
var memoryValues = [];
var tileId = [];
var flippedTiles = 0;
var level = 0;
var moves = 0;
var noOfTilesInGridRow;
var currentArraylength;
var interval;
var second = 0;
var minute = 0;
var hour = 0;
var delay;
var currentTime;
var remark;
var totalMoves = 0;
var totalArrayLength = 0;

// This function is used for shuffling the array elemenets
Array.prototype.ShuffleTiles = function()
{
  var i = currentArraylength, j, temp;
  while(--i > 0)
  {
    j = Math.floor(Math.random() * (i+1));
    temp = this[j];
    this[j] = this[i];
    this[i] = temp;
  }
}


// This is the game Timer function
function starttimmer()
{
  var timmer = document.querySelector(".timmer");
  interval = setInterval(function(){
    currentTime = timmer.innerHTML = "Timer: "+minute+" mins "+second+" secs";
    second++;
    if(second == 60)
    {
      minute++;
      second=0;
    }
    if(minute == 60)
    {
      hour++;
      minute = 0;
    }
  },1000);
}

//This is the function to track moves and change star rating
function movesCounter()
{
  moves++;
  if(flippedTiles==currentArraylength)
  {
    //tracking of total moves in game
    totalMoves = totalMoves + moves;
  }
  //Starts the timer when the first move is made
  if(moves == 1)
  {
    //remark to be displayed after the level
    remarkText = "Excellent Memory!";
    starttimmer();
  }
  /* The follwing statements check if the moves made to comlete the game are in range are defined value.
  The defined values are in proportion 1/2,1 and 2/3 greater than total tiles for each star respectively.
  Changes the star rating depending on the moves
  It also decides the remarks based on the no.of moves made to finish the game*/
  if(moves == (currentArraylength+(currentArraylength/2)+1))
  {
    remarkText = "Good Memomry!";
    var star3Div = document.getElementById("star3");
    star3Div.classList.remove("showStar");
    star3Div.classList.add("hideStar");
  }
  if(moves == ((currentArraylength*2)+1))
  {
    remarkText = "Ok memory!";
    var star2Div = document.getElementById("star2");
    star2Div.classList.remove("showStar");
    star2Div.classList.add("hideStar");
  }
  if(moves >= (currentArraylength+((currentArraylength*3)/2)+1))
  {
    remarkText = "Try Hard";
    var star1Div = document.getElementById("star1");
    star1Div.classList.remove("showStar");
    star1Div.classList.add("hideStar");
  }
  //add current moves to the screen
  var movesDiv = document.getElementById("movesId");
  movesDiv.innerHTML = "Moves: "+moves;
}

//This sequence runs at the end of each level
function levelEndSequence()
{
  totalArrayLength = totalArrayLength + currentArraylength;
  //check if last level is reached, if not than generate the replay or next level sequence
  if(level < 14){
    //freezes the timer
    var timmer = document.querySelector(".timmer");
    timmer.innerHTML = currentTime;
    clearInterval(interval);
    //resets values
    flippedTiles = 0;
    second = 0;
    minute = 0;
    hour = 0;
    moves = 0;

    //generate the end level buttons and text.
    //Places all the buttons inside a sequence container
    var gameBoardDiv = document.querySelector(".gameBoardDiv");
    gameBoardDiv.innerHTML = "";

    var endLevelText = createParagraph("Level "+level+" Completed");
    endLevelText.classList.add("endLevelText");
    var remark = createParagraph(remarkText);
    remark.classList.add("remark");
    var sequenceContainer = createDiv();
    sequenceContainer.classList.add("sequenceContainer");
    var buttonDiv = createDiv();
    var replayButton = createButton("Replay Level");
    replayButton.onclick = game;
    var nextLevelButton = createButton("Next Level");
    nextLevelButton.onclick = levelSelector;
    buttonDiv.classList.add("buttonDiv");
    replayButton.classList.add("buttonLeft");
    nextLevelButton.classList.add("buttonRight");
    sequenceContainer.appendChild(remark);
    sequenceContainer.appendChild(endLevelText);
    buttonDiv.appendChild(replayButton);
    buttonDiv.appendChild(nextLevelButton);
    sequenceContainer.appendChild(buttonDiv);
    gameBoardDiv.appendChild(sequenceContainer);
  }

  //if the last level is reached than generates the end game board
  else
  {
    //congratulations audio
    var winAudio = createAudio("Audio/5_Sec_Crowd_Cheer-Mike_Koenig-1562033255.wav");
    winAudio.play();

    //clears the screen
    document.body.innerHTML = "";
    var winDiv = createDiv();

    //Checks the total no.of moves made for the completion of game and decides the player status.
    if (totalMoves <= (totalArrayLength+(totalArrayLength/2))){
      window.finalRemark = "Excelent Memory!";
    }
    else if (totalMoves <= (totalArrayLength+(totalArrayLength/2))){
      finalRemark = "Good Memory!";
    }
    else if (totalMoves <= (totalArrayLength+((totalArrayLength*3)/2))){
      finalRemark = "Nice Memory!";
    }
    else{
      finalRemark = "Try Hard!";
    }

    //container to hold the win sequence
    winDiv.classList.add("winContainer");
    var winText1 = createParagraph("Congratulation!");
    var winText3 = createParagraph(finalRemark);
    var winText2 = createParagraph("You have Completed the game in " + totalMoves + " Moves and Matched " + totalArrayLength +" tiles");
    winText1.classList.add("win1");
    winText2.classList.add("win2");
    winText3.classList.add("win3");
    winDiv.appendChild(winText1);
    winDiv.appendChild(winText2);
    winDiv.appendChild(winText3);
    document.body.appendChild(winDiv);
  }
}

//the below function flips the tiles and also checks if they are same
function flipTiles(tile,value)
{
  //the below classes are toggled to reveal and hide the tiles
  let front = "front"+tile.id;
  let back = "back"+tile.id;
  let tileFront = document.getElementById(front);
  let tileBack = document.getElementById(back);
  tileFront.classList.remove("front");
  tileBack.classList.add("back");
  //checks conditions for matching the tile
  //if the tile is already matched and if the reference array has only 2 values
  if(tileBack.innerHTML == "" && memoryValues.length < 2)
  {
    var selectAudio = createAudio("Audio/Pop Clip In-SoundBible.com-583746573.wav");
    selectAudio.play();
    //the first clicked tile name is stored in the reference array
    if(memoryValues.length == 0)
    {
      memoryValues.push(value.name);
      tileId.push(tile.id);
      //the second clicked tile name is stored in the reference array
    } else if(memoryValues.length == 1)
    {
      memoryValues.push(value.name);
      tileId.push(tile.id);
      //checks if tiles are the same
      if(memoryValues[0] == memoryValues[1])
      {
        setTimeout(function()
        {
          var correctAudio = createAudio("Audio/Mario_Jumping-Mike_Koenig-989896458.wav");
          correctAudio.play();
        },500);
        flippedTiles += 2;
        // Clear both arrays
        memoryValues = [];
        tileId = [];
        // If the game board tiles are matched completed levelEndSequence is called
        setTimeout(function ()
        {
          if(flippedTiles == currentArraylength)
          {
            levelEndSequence();
          }
        },600);
      }
      else {
        function flipBack()
        {
          var incorrectAudio = createAudio("Audio/Strong_Punch-Mike_Koenig-574430706.wav");
          incorrectAudio.play();
          // Flip the 2 tiles back over
          var tile1Back = document.getElementById("back"+tileId[0]);
          var tile2Back = document.getElementById("back"+tileId[1]);
          var tile1Front = document.getElementById("front"+tileId[0]);
          var tile2Front = document.getElementById("front"+tileId[1]);
          tile1Back.innerHTML = "";
          tile2Back.innerHTML = "";
          tile1Front.classList.toggle("front");
          tile2Front.classList.toggle("front");
          tile1Back.classList.toggle("back");
          tile2Back.classList.toggle("back");
          // Clear both arrays
          memoryValues = [];
          tileId = [];
        }
        //delays flipback function so that the second tiles is revealed for some time
        setTimeout(flipBack,500);
      }
    }
    movesCounter();
  }
}

//This function will create the game area and calls the necessary functions to run the game
function game()
{
  //clears HTML before generating game
  document.body.innerHTML="";
  //creting game heading
  createHeading("Memory Game");
  //Score Elements are created and are placed in a container
  var scoreElements = createDiv();
  //stars section and stars are created
  var star1Div = createDiv();
  star1Div.id = "star1";
  star1Div.classList.add("showStar");
  var star2Div = createDiv();
  star2Div.id = "star2";
  star2Div.classList.add("showStar");
  var star3Div = createDiv();
  star3Div.id = "star3";
  star3Div.classList.add("showStar");
  var starsDiv = createDiv();
  starsDiv.classList.add("starsContainer");
  starsDiv.appendChild(star1Div);
  starsDiv.appendChild(star2Div);
  starsDiv.appendChild(star3Div);
  //timer section is created
  var timmerDiv = createDiv();
  timmerDiv.classList.add("timmer");
  timmerDiv.innerHTML = "Timer: 0 mins 0 secs";
  //moves section is created
  var movesDiv = createDiv();
  movesDiv.classList.add("moves");
  movesDiv.id = "movesId";
  movesDiv.innerHTML = "Moves: 0"

  scoreElements.classList.add("scoreElementsContainer")
  scoreElements.appendChild(starsDiv);
  scoreElements.appendChild(timmerDiv);
  scoreElements.appendChild(movesDiv);
  document.body.appendChild(scoreElements);


  //creates game board
  var gameBoardDiv = createDiv();
  gameBoardDiv.classList.add("gameBoardDiv");
  if (window.matchMedia('screen and (max-width: 825px)').matches) {
    window.divWidth = (70)*(noOfTilesInGridRow);
    gameBoardDiv.style.width = divWidth+"px";
    document.body.appendChild(gameBoardDiv);
  }
  else {
    window.divWidth = (140)*(noOfTilesInGridRow);
    gameBoardDiv.style.width = divWidth+"px";
    document.body.appendChild(gameBoardDiv);
  }
  //assigning width to game board depending on array length


  //calling the shuffle function
  memoryTilesArray.ShuffleTiles();

  //creating tiles for the game
  for(let i = 0; i < currentArraylength; i++)
  {
    var tiles = createDiv();
    tiles.classList.add("tile");
    tiles.id = i;
    let tileFront = createDiv();
    tileFront.id = "front"+i;
    let tileBack = createDiv();
    let val = memoryTilesArray[i];
    //tile is assigned with the value which will be visible and revealed at the starting of the game
    tileBack.style.background = "url("+ val.img +") no-repeat center center";
    tileBack.style.backgroundSize = "100%";
    tileBack.classList.add("back");
    tileBack.id = "back"+i;
    //reveals the tiles and hides them for defined time for each level
    setTimeout(function()
    {
      tileBack.classList.remove("back");
      tileFront.classList.add("front");
    },delay);
    //adds front and back of the cards
    tiles.appendChild(tileFront);
    tiles.appendChild(tileBack);
    //Calling the function flipTiles when a tile is clicked
    tiles.addEventListener('click', function()
    {
      flipTiles(this,val);
    });
    gameBoardDiv.appendChild(tiles);
  }
}

//This function increases the levels after each game and defines the values for each level of the game from level 1 to level 14
function levelSelector()
{
  level ++;
  if(level == 1){
    memoryTilesArray = gameTiles[0];
    currentArraylength = 4;
    noOfTilesInGridRow = 2;
    delay = 1000;
    game();
  }
  if(level == 2)
  {
    memoryTilesArray = gameTiles[1];
    currentArraylength = 4;
    noOfTilesInGridRow = 2;
    delay = 1200;
    game();
  }
  if(level == 3)
  {
    memoryTilesArray = gameTiles[0];
    currentArraylength = 6;
    noOfTilesInGridRow = 3;
    delay = 1500;
    game();
  }
  if(level == 4)
  {
    memoryTilesArray = gameTiles[1];
    currentArraylength = 6;
    noOfTilesInGridRow = 3;
    delay = 1700;
    game();
  }
  if(level == 5)
  {
    memoryTilesArray = gameTiles[0];
    currentArraylength = 8;
    noOfTilesInGridRow = 4;
    delay = 1900;
    game();
  }
  if(level == 6)
  {
    memoryTilesArray = gameTiles[1];
    currentArraylength = 8;
    noOfTilesInGridRow = 4;
    delay = 2500;
    game();
  }
  if(level == 7)
  {
    memoryTilesArray = gameTiles[0];
    currentArraylength = 12;
    noOfTilesInGridRow = 4;
    delay = 2600;
    game();
  }
  if(level == 8)
  {
    memoryTilesArray = gameTiles[1];
    currentArraylength = 12;
    noOfTilesInGridRow = 4;
    delay = 2800;
    game();
  }
  if(level == 9)
  {
    memoryTilesArray = gameTiles[0];
    currentArraylength = 16;
    noOfTilesInGridRow = 4;
    delay = 3900;
    game();
  }
  if(level == 10)
  {
    memoryTilesArray = gameTiles[1];
    currentArraylength = 16;
    noOfTilesInGridRow = 4;
    delay = 4200;
    game();
  }
  if(level == 11)
  {
    memoryTilesArray = gameTiles[0];
    currentArraylength = 20;
    noOfTilesInGridRow = 5;
    delay = 4400;
    game();
  }
  if(level == 12)
  {
    memoryTilesArray = gameTiles[1];
    currentArraylength = 20;
    noOfTilesInGridRow = 5;
    delay = 4700;
    game();
  }
  if(level == 13)
  {
    memoryTilesArray = gameTiles[0];
    currentArraylength = 24;
    noOfTilesInGridRow = 6;
    delay = 5000;
    game();
  }
  if(level == 14)
  {
    memoryTilesArray = gameTiles[1];
    currentArraylength = 24;
    noOfTilesInGridRow = 6;
    delay = 5400;
    game();
  }
}
//starts the function game when the HTML page loads
window.onload = levelSelector;
